### PR TITLE
update UpsertProductCategory logic to fix upsert issue when try to hi…

### PR DESCRIPTION
**Pull request by nurcholisnanda**
…t update api

### CODE REVIEW

**Review by arnesuscharles — APPROVED** [review]

File: service/service.go [hunk 1]
```diff
@@ -407,10 +407,20 @@ func (s *service) UploadImageToStorage(ctx context.Context, imageBase64Str strin
 	return imageId, nil
 }
 
-func (s *service) UpsertProductCategory(ctx context.Context, prodCategory *prodEntity.ProductCategory) error {
-	existingProdCategory, err := s.productRepository.GetProductCategoryByName(ctx, strings.ToLower(prodCategory.Name))
-	if err != nil {
-		return err
+func (s *service) UpsertProductCategory(ctx context.Context, prodCategory *prodEntity.ProductCategory) (err error) {
+	var existingProdCategory *prodEntity.ProductCategory
+	switch prodCategory.ID {
```
Comment (l200180102-student) [inline service/service.go:412]: how many categories do we have for the product mas?
and if we use **switch case**, it is better if we have a specific value for each category.
 I think we can use if instead:
example:
```
if prodCategory.ID == 0 {
   // get product by name
} else {
   // get product by id
}
```

File: service/service.go [hunk 2]
```diff
@@ -407,10 +407,20 @@ func (s *service) UploadImageToStorage(ctx context.Context, imageBase64Str strin
...
+		existingProdCategory, err = s.productRepository.GetProductCategoryByName(ctx, strings.ToLower(prodCategory.Name))
+		if err != nil {
+			return err
+		}
+	default:
+		if prodCat, err := s.productRepository.GetProductCategoryById(ctx, prodCategory.ID); err != nil {
+			return status.Errorf(codes.Internal, "Error getting product category by id data")
+		} else if prodCat == nil {
+			return status.Errorf(codes.NotFound, "Product category id is not found")
+		}
```
Comment (paulusherdiyatma) [inline service/service.go]: how many categories do we have for the product mas?
and if we use **switch case**, it is better if we have a specific value for each category.
 I think we can use if instead:
example:
```
if prodCategory.ID == 0 {
   // get product by name
} else {
   // get product by id
}
```

Comment (nurcholisnanda) [reply service/service.go]: Hi mas @paulusherdiyatma ,
I have updated the code based on our discussion
please re-review it

File: service/constant.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+type CodeDetail string
+
+const (
+	ErrNotFound = "not found"
+
+	//Insert Error_Detail here
+	ERR_RECORD_NOT_FOUND CodeDetail = "ERR_RECORD_NOT_FOUND"
+	ERR_INTERNAL_ERROR   CodeDetail = "ERR_INTERNAL_ERROR"
+	ERR_RECORD_IS_EXIST  CodeDetail = "ERR_RECORD_IS_EXIST"
```
Comment (paulusherdiyatma) [inline service/constant.go]:
```suggestion
	ERR_RECORD_NOT_FOUND CodeDetail = "record is not found"
	ERR_INTERNAL_SERVER   CodeDetail = "internal server error"
	ERR_RECORD_IS_EXIST  CodeDetail = "record already exist"
```

**Review by paulusherdiyatma — APPROVED** [review]
LGTM

**Review by arnesuscharles — APPROVED** [review]

**Pull request merged**